### PR TITLE
[EventHubs] add kind to amqp transport files

### PR DESCRIPTION
**Pull request by swathipil**
Currently, we're checking which AmqpTransport is being used with isinstance/other attributes. Instead, we should follow the SB pattern and add `KIND` to the transports to check.

### CODE REVIEW

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer.py [hunk 1]
```diff
@@ -183,8 +184,8 @@ def _next_message_in_buffer(self):
         # pylint:disable=protected-access
         message = self._message_buffer.popleft()
         event_data = EventData._from_message(message)
-        if self._amqp_transport != PyamqpTransport:
-            event_data._uamqp_message == message    # pylint: disable=pointless-statement
+        if self._amqp_transport.KIND != "pyamqp":
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_consumer.py:187]: This seems a bit more readible:
```suggestion
        if self._amqp_transport.KIND == "uamqp":
```

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer.py [hunk 2]
```diff
@@ -183,8 +184,8 @@ def _next_message_in_buffer(self):
...
+            event_data._uamqp_message = message    # pylint: disable=pointless-statement
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_consumer.py:188]: I think the pylint disable can be removed now that you've fixed the `==`
```suggestion
            event_data._uamqp_message = message
```

File: sdk/eventhub/azure-eventhub/azure/eventhub/aio/_transport/_base_async.py [hunk 3]
```diff
@@ -17,6 +17,9 @@ class AmqpTransportAsync(ABC):  # pylint: disable=too-many-public-methods
     """
     Abstract class that defines a set of common methods needed by producer and consumer.
     """
+
+    KIND = "uamqp"
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/azure/eventhub/aio/_transport/_base_async.py:21]: Why are we setting "uamqp" on the ABC?

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer.py (hunk 2 above)
Comment (swathipil) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_consumer.py:188]: oo yep makes sense!

File: sdk/eventhub/azure-eventhub/azure/eventhub/aio/_transport/_base_async.py (hunk 3 above)
Comment (swathipil) [reply sdk/eventhub/azure-eventhub/azure/eventhub/aio/_transport/_base_async.py:21]: oops sorry fixed!

**Review by kashifkhan — APPROVED** [review]

**Pull request merged**